File: typescript/linkedList.ts
namespace NSLinkedList {
    export class Node {
        value: number;
        next: Node | null = null;

        constructor(value: number) {
            this.value = value;
        }
    }

    export class LinkedList {
        private initial: Node | null = null;
        private size: number = 0;

        private getNode(index: number) {
            const _index = index < 0 ? this.size + index : index;
            let node = this.initial;
            for (let i = 0; i < _index; i++) {
                if (node !== null) {
                    node = node.next;
                } else {
                    break;
                }
            }
            if (node !== null && !(_index < 0)) {
                return node;
            }
            throw new RangeError();
        }

        get(index: number) {
            const node = this.getNode(index);
            return node.value;
        }

        add(value: number): void {
            const newNode = new Node(value);
            if (this.size === 0) {
                this.initial = newNode;
            } else {
                let lastNode = this.initial as Node;
                while (lastNode.next !== null) {
                    lastNode = lastNode.next;
                }
                lastNode.next = newNode;
            }
            this.size++;
        }

        remove(value: number) {
            if (this.size === 0) {
                throw new Error();
            }
            if (this.initial !== null) {
                if (this.initial.value === value) {
                    let aux = this.initial.next;
                    this.initial.next = null;
                    this.initial = aux;
                    this.size--;
                    return true;
                } else {
                    let previousNode = this.initial;
                    let currentNode = previousNode.next;
                    while (currentNode !== null) {
                        if (currentNode.value === value) {
                            previousNode.next = currentNode.next;
                            currentNode.next = null;
                            this.size--;
                            return true;
                        }
                        previousNode = currentNode;
                        currentNode = currentNode.next;
                    }
                    return false;
                }
            }
        }

        get length(): number {
            return this.size;
        }
    }
}

File: python/linked_list.py
from typing import TypeVar

TNode = TypeVar("TNode", bound="Node")


class Node:
    value: int
    next: TNode | None = None

    def __init__(self, value: int):
        self.value = value


class LinkedList:
    _initial: Node | None = None
    _size: int = 0

    def _get_node(self, index: int):
        _index = index
        if index < 0:
            _index = self._size + index
            if _index < 0:
                raise IndexError()
        node = self._initial
        for _ in range(_index):
            if node is None:
                raise IndexError()
            node = node.next
        return node

    def add(self, value: int):
        new_node = Node(value)
        if self._size == 0:
            self._initial = new_node
        else:
            last_node: Node = self._initial
            while last_node.next is not None:
                last_node = last_node.next
            last_node.next = new_node
        self._size += 1

    def remove(self, value: int):
        if self._size == 0:
            return ValueError()
        if self._initial.value == value:
            self._initial = self._initial.next
            self._size -= 1
            return True
        else:
            previous_node = self._initial
            current_node = previous_node.next
            while current_node is not None:
                if current_node.value == value:
                    previous_node.next = current_node.next
                    current_node.next = None
                    self._size -= 1
                    return True
                previous_node = current_node
                current_node = current_node.next
            return False

    def __len__(self):
        return self._size

    def __getitem__(self, key: int):
        if self._size == 0:
            raise IndexError()
        node = self._get_node(key)
        return node.value
